scripts/test/Muon/fitting_widgets/basic_fitting/basic_fitting_presenter_test.py

```diff
@@ -0,0 +1,512 @@
+        self.assertEqual(self.view.set_slot_for_use_raw_changed.call_count, 1)
+
+    def tearDown(self):
+        self.presenter = None
+        self.model = None
+        self.view = None
+
+    def test_that_handle_ads_clear_or_remove_workspace_event_will_attempt_to_reset_all_the_data_and_enable_gui(self):
+        self.presenter.handle_ads_clear_or_remove_workspace_event()
+
+        self.presenter.update_and_reset_all_data.assert_called_with()
+        self.presenter.enable_editing_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_handle_ads_clear_or_remove_workspace_event_will_disable_the_tab_if_no_data_is_loaded(self):
+        self.mock_model_number_of_datasets = mock.PropertyMock(return_value=0)
+        type(self.model).number_of_datasets = self.mock_model_number_of_datasets
+
+        self.presenter.handle_ads_clear_or_remove_workspace_event()
+
+        self.presenter.update_and_reset_all_data.assert_called_with()
+        self.presenter.reset_tab_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_handle_gui_changes_made_will_reset_the_start_and_end_x_in_the_model_and_view(self):
+        self.presenter.handle_gui_changes_made({"FirstGoodDataFromFile": True})
+
+        self.model.reset_start_xs_and_end_xs.assert_called_once_with()
+        self.mock_model_current_start_x.assert_called_once_with()
+        self.mock_view_start_x.assert_called_once_with(self.start_x)
+        self.mock_model_current_end_x.assert_called_once_with()
+        self.mock_view_end_x.assert_called_once_with(self.end_x)
+
+    def test_that_handle_new_data_loaded_will_attempt_to_reset_all_the_data_and_enable_the_gui(self):
+        self.presenter.clear_cached_fit_functions = mock.Mock()
+
+        self.presenter.handle_new_data_loaded()
+
+        self.presenter.update_and_reset_all_data.assert_called_with()
+        self.mock_view_plot_guess.assert_called_once_with(False)
+        self.presenter.clear_cached_fit_functions.assert_called_with()
+        self.presenter.enable_editing_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_handle_new_data_loaded_will_disable_the_tab_if_no_data_is_loaded(self):
+        self.mock_model_number_of_datasets = mock.PropertyMock(return_value=0)
+        type(self.model).number_of_datasets = self.mock_model_number_of_datasets
+        self.presenter.clear_cached_fit_functions = mock.Mock()
+
+        self.presenter.handle_new_data_loaded()
+
+        self.presenter.update_and_reset_all_data.assert_called_with()
+        self.mock_view_plot_guess.assert_called_once_with(False)
+        self.presenter.clear_cached_fit_functions.assert_called_with()
+        self.presenter.reset_tab_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_handle_plot_guess_changed_will_update_plot_guess_using_the_model(self):
+        self.presenter.handle_plot_guess_changed()
```

Comment:
does this need the plot_guessto be set?

Reply:
Yes, plot guess is set in the setup of this test